xc/device.ml

```diff
@@ -607,12 +607,33 @@ let qemu_media_change ~xs device _type params =
 
 	if is_upstream_qemu device.frontend.domid
 	then begin
+		let open Qmp in
 		let cd = "ide1-cd1" in
-		let qmp_cmd =
-			if params = ""
-			then (Qmp.Command(None, Qmp.Eject (cd, Some true)))
-			else (Qmp.Command(None, Qmp.Change (cd, params, None))) in
-		qmp_write device.frontend.domid qmp_cmd
+		match params with
+		| "" ->
+			let qmp_cmd = Command(None, Eject (cd, Some true)) in
+			qmp_write device.frontend.domid qmp_cmd
+		| _ ->
+			let c = Qmp_protocol.connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" device.frontend.domid) in
+			let fd_of_cd = Unix.openfile params [ Unix.O_RDONLY ] 0o640 in
```

Comment:
Thanks for your comments. 
Updated the code, I think it may have below exceptions:
```
1. Fail to connect socket
2. Fail to open CD Image
```
The code cannot continue with these exceptions, so I just raise it out.